spark-3.3/clickhouse-spark/src/main/scala/com/clickhouse/spark/ClickHouseTableProvider.scala

```diff
@@ -0,0 +1,215 @@
+
+    // Infer schema from table using catalog
+    val catalog = createCatalog(options)
+    val identifier = extractIdentifier(options)
+    try {
+      val table = catalog.loadTable(identifier)
+      log.info(s"Inferred schema from table with ${table.schema.fields.length} fields")
+      table.schema
+    } finally {
+      // Catalog manages its own resources via NodeClient
+    }
+  }
+
+  override def getTable(
+    schema: StructType,
+    partitioning: Array[Transform],
+    properties: util.Map[String, String]
+  ): Table = {
+    val options = new CaseInsensitiveStringMap(properties)
+    log.info(s"Getting ClickHouse table with schema: ${schema.simpleString}")
+
+    val catalog = createCatalog(options)
+    val identifier = extractIdentifier(options)
+
+    loadOrCreateTable(catalog, identifier, schema, partitioning, properties)
+  }
+
+  private def loadOrCreateTable(
+    catalog: ClickHouseCatalog,
+    identifier: Identifier,
+    schema: StructType,
+    partitioning: Array[Transform],
+    properties: util.Map[String, String]
+  ): Table =
+    try
+      catalog.loadTable(identifier)
+    catch {
+      case _: NoSuchTableException =>
+        log.info(s"Table $identifier does not exist, creating with provided schema")
+        createTableWithOrderBy(catalog, identifier, schema, partitioning, properties)
+    }
+
+  private def createTableWithOrderBy(
+    catalog: ClickHouseCatalog,
+    identifier: Identifier,
+    schema: StructType,
+    partitioning: Array[Transform],
+    properties: util.Map[String, String]
+  ): Table = {
+    import scala.collection.JavaConverters._
+    val propsMap = properties.asScala.toMap
+    val updatedProps = ensureOrderByConfiguration(schema, propsMap)
+    catalog.createTable(identifier, schema, partitioning, updatedProps.asJava)
+  }
+
+  private def ensureOrderByConfiguration(schema: StructType, props: Map[String, String]): Map[String, String] = {
+    val hasOrderBy = props.contains("order_by") || props.contains("local.order_by")
+
+    if (!hasOrderBy) {
+      addDefaultOrderBy(schema, props)
+    } else {
+      ensureNullableKeySupport(schema, props)
+    }
+  }
+
+  private def addDefaultOrderBy(schema: StructType, props: Map[String, String]): Map[String, String] = {
+    log.debug(s"Schema fields: ${schema.fields.map(f => s"${f.name}(nullable=${f.nullable})").mkString(", ")}")
+
+    val orderByColumn = schema.fields
+      .find(!_.nullable)
+      .map(_.name)
+      .getOrElse {
+        throw new IllegalArgumentException(
+          "Cannot auto-generate ORDER BY clause: all columns are nullable. " +
+            "ClickHouse Cloud requires non-nullable columns in ORDER BY. " +
+            "Please either: (1) make at least one column non-nullable, or " +
+            "(2) explicitly set 'order_by' option to specify which column(s) to use."
+        )
+      }
+
+    log.info(s"Auto-generated ORDER BY clause: $orderByColumn")
```

Comment:
I'm not sure if such an automatic behavior is a good idea, could you please elaborate more about your consideration? and this at least requires a config to allow the user to disable it.